eng/packages/http-client-csharp/generator/Azure.Generator/src/Visitors/DistributedTracingVisitor.cs

```diff
@@ -0,0 +1,264 @@
+            }
+
+            return constructor;
+        }
+
+        protected override MethodProvider? VisitMethod(MethodProvider method)
+        {
+            if (ShouldSkip(method.EnclosingType) || !IsSubClientFactoryMethod(method))
+            {
+                return base.VisitMethod(method);
+            }
+
+            UpdateDistributedTracingRefInSubClientFactoryMethod(method);
+
+            return method;
+        }
+
+        protected override ScmMethodProvider? VisitMethod(ScmMethodProvider method)
+        {
+            if (ShouldSkip(method.EnclosingType) || !method.IsProtocolMethod)
+            {
+                return base.VisitMethod(method);
+            }
+
+            // Wrap protocol methods with distributed tracing
+            UpdateProtocolMethodsWithDistributedTracing(method);
+
+            return method;
+        }
+
+        private static void UpdateDistributedTracingRefInSubClientFactoryMethod(
+            MethodProvider method)
+        {
+            if (method.BodyStatements != null)
+            {
+                PropertyProvider clientDiagnosticsProperty = method.EnclosingType.Properties
+                    .First(p => p.Name == ClientDiagnosticsPropertyName);
+                List<MethodBodyStatement> updatedFactoryMethodStatements = [];
+                foreach (var statement in method.BodyStatements.Flatten())
+                {
+                    if (TryUpdateSubClientFactoryMethodReturnStatement(
+                        statement,
+                        clientDiagnosticsProperty,
+                        out MethodBodyStatement? updatedStatement))
+                    {
+                        updatedFactoryMethodStatements.Add(updatedStatement);
+                    }
+                    else
+                    {
+                        updatedFactoryMethodStatements.Add(statement);
+                    }
+                }
+
+                method.Update(bodyStatements: updatedFactoryMethodStatements);
+            }
+        }
+
+        private static void UpdateProtocolMethodsWithDistributedTracing(MethodProvider method)
+        {
+            string scopeName = $"{method.EnclosingType.Name}.{method.Signature.Name}";
+            PropertyProvider clientDiagnosticsProperty = method.EnclosingType.Properties.First(p => p.Name == ClientDiagnosticsPropertyName);
+
+            // declare scope
+            var scopeDeclaration = UsingDeclare(
+                "scope",
+                typeof(DiagnosticScope),
+                clientDiagnosticsProperty.Invoke(nameof(ClientDiagnostics.CreateScope), [Literal(scopeName)], false, false),
+                out var scope);
+            // start scope
+            var scopeStart = scope.Invoke(nameof(DiagnosticScope.Start)).Terminate();
+            // wrap existing statements in try / catch
+            var tryStatement = new TryStatement
+                {
+                    method.BodyStatements ?? MethodBodyStatement.EmptyLine
```

Comment:
I think if there are no body statements that means there is a BodyExpression. I don't think this is actually possible right now, but we may need to wrap the expression instead.